DQM/EcalCommon/src/DQWorker.cc

```diff
@@ -115,39 +115,63 @@ namespace ecaldqm {
     edso_.topology = &_es.getData(topoHandleEndLumi);
   }
 
-  EcalElectronicsMapping const *DQWorker::GetElectronicsMap() {
-    if (!edso_.electronicsMap)
+  bool DQWorker::checkElectronicsMap(bool doThrow /* = true*/) {
+    if (edso_.electronicsMap)
+      return true;
+    if (doThrow)
       throw cms::Exception("InvalidCall") << "Electronics Mapping not initialized";
+    return false;
+  }
+
+  bool DQWorker::checkTrigTowerMap(bool doThrow /* = true*/) {
+    if (edso_.trigtowerMap)
+      return true;
+    if (doThrow)
+      throw cms::Exception("InvalidCall") << "TrigTowerConstituentsMap not initialized";
+    return false;
+  }
+
+  bool DQWorker::checkGeometry(bool doThrow /* = true*/) {
+    if (edso_.geometry)
+      return true;
+    if (doThrow)
+      throw cms::Exception("InvalidCall") << "CaloGeometry not initialized";
+    return false;
+  }
+
+  bool DQWorker::checkTopology(bool doThrow /* = true*/) {
+    if (edso_.topology)
+      return true;
+    if (doThrow)
+      throw cms::Exception("InvalidCall") << "CaloTopology not initialized";
+    return false;
+  }
+
+  EcalElectronicsMapping const *DQWorker::GetElectronicsMap() {
+    checkElectronicsMap();
```

Comment:
The argument `doThrow` is not initialized: which value is selected here?
Shouldn't you call it with a `true` argument, or alternatively initialize doThrow inside the checkXXX methods?

Reply:
The argument is set to `true` by default here: https://github.com/cms-sw/cmssw/pull/38503/files/bece5b88b06f17ce1c6dd3c89c2715f384374862#diff-5996ad1e366121e3ee83866beb5b0ecddeea3cfbdf0d48e3d2041f3a9d860670R98-R101
so passing no argument is the same as passing `true`

Reply:
Indeed... Thank yu @alejands